### DISCUSSION TITLE
swagger-aggregator-config configmap includes COMMIT_SHA is needed variable to interpolate

### DISCUSSION THREAD

**Discussion by phil-hotchkiss** [mia-platform/community#389] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

12.1.2

### Description

My company uses the Enhanced Workflow and pull based deployment model. When using the swagger-aggregator, the configmap named swagger-aggregator-config includes a variable that would need interpolation, but doesn't make sense as an environment variable. In the imperative deployment model, this value seems like it would be perfectly appropriate, but it doesn't seem to meet the pull based deployment use case. As you can see in the attached image, the value of this configmap includes {{COMMIT_SHA}} referenced as the value of `version`

![image](https://github.com/mia-platform/community/assets/136851567/d3758d93-7bb8-4585-9039-23abe38e9879)

Please let me know if there are any suggestions for how to work around this for the time being. I could always set a bogus value to COMMIT_SHA and get this to work, but I am assuming that's not the intent and not entirely sure if that would have any negative consequences or side effects. 

As an additional note, if this is a common pattern among marketplace items, it may be worth combing through them and deciding if there is a more appropriate value to use than the COMMIT_SHA in marketplace components. 

### Actual Outcome

{{COMMIT_SHA}} included as version value, but this value doesn't make a lot of sense for pull based deployments. 

### Expected Outcome

For pull based deployment models (or maybe all deployment models), replace the version value with something that has control over this specific value.

**Comment by fredmaggiowski** [comment, marked as answer]
Hi @phil-hotchkiss sorry for the inconvenience, this is definitely an issue with the enhanced workflow, we are planning a fix for this in the next release. 

For the time being you should be able to add a public variable with the COMMIT_SHA key (the value is not important) so that it gets interpolated and the problem should disappear

The purpose for the interpolation in that config is to set metadata in the generated openapi spec, so nothing of vital importance

**Reply by phil-hotchkiss** [reply]
Sounds good. We'll proceed with that as the expectation for now. Thanks

**Reply by phil-hotchkiss** [reply]
@fredmaggiowski For what it's worth, I ran into the same issue with {{STAGE_TO_DEPLOY}} in multiple microservices.

**Reply by fredmaggiowski** [reply]
STAGE_TO_DEPLOY is a managed variable so it shouldn't give problems. That's weird, let me have a look and come back to you later.

In the meantime you should still be able to temporarily set it with a public variable.

Can you share an example service configuration that is affected by this problem (either the json or a screenshot of the design section should be enough) so that I can try and set up a repro case? Thanks!
